src/lib/utils/atomic_max.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "storage/mvcc_data.hpp"
+
+namespace hyrise {
+
+// std::max()-like assignment for std::atomic<T>. Taken from https://stackoverflow.com/a/16190791, which is a slightly
+// adapted version of https://herbsutter.com/2012/08/31/reader-qa-how-to-write-a-cas-loop-using-stdatomics/.
+template <typename T>
+inline void set_atomic_max(std::atomic<T>& maximum_value, const T& value) noexcept {
+  auto prev_value = maximum_value.load();
+  while (prev_value < value && !maximum_value.compare_exchange_weak(prev_value, value)) {}
+}
+
+// Specialization for CommitID: Ignore MAX_COMMIT_ID as reserved value.
```

Comment:
Can you elaborate what the intention is of comparing `prev_value` with `MvccData::MAX_COMMIT_ID`?

Reply:
Added the comment from the test.